core/arch/arm/plat-imx/conf.mk

```diff
@@ -80,6 +80,9 @@ endif
 
 ifeq ($(filter y, $(CFG_PSCI_ARM32)), y)
 CFG_HWSUPP_MEM_PERM_WXN = n
+CFG_IMX_WDOG ?= y
 endif
 
+MAX_MMAP_REGIONS ?= 24
```

Comment:
This variable should have a `CFG_` prefix, also the config magic is only applied to `CFG_` variables if I'm not misstaken.

Reply:
Per @etienne-lms 's comments:
```
I think the platform_config.h could set MAX_MMAP_REGIONS and change here:

+#ifndef MAX_MMAP_REGIONS
 #define MAX_MMAP_REGIONS	13
+#endif
Main point is that this config information does not really need to be CFG_xxx format which are mainly used for config settings required in makefile scripts and eventually also source files.
```
I changed to use MAX_MMAP_REGIONS.

So @jenswi-linaro @jforissier, you agree to use the following, add platform could define CFG_MMAP_REGIONS?
```
#ifndef CFG_MMAP_REGIONS
#define CFG_MMAP_REGIONS 13
#endif
#define MAX_MMAP_REGIONS CFG_MMAP_REGIONS
```

Reply:
First thing: generally speaking, `#ifndef CFG_X / #define CFG_X / #endif` is not good. `CFG_x ?= some_value` in some `.mk` file should always be preferred.

Second thing: whether or not the variable should be a `CFG_` is easy to decide. If it makes sense to choose the value a compile time, use CFG_. Otherwise, hard-code the value in a .h and do NOT use #ifndef.

Third thing: I don't understand why you would need two variables (MAX_MMAP_REGIONS and CFG_MMAP_REGIONS).

It seems to me `CFG_MMAP_REGIONS ?= some_value` is all we need. It may be set globally in `core/config.mk` and a platform-specific value may be given in `core/arch/arm/plat-imx/conf.mk`. A compile-time assert may be in order to make sure the value is acceptable in the end.

Reply:
Thanks @jforissier . Replacing MAX_MMAP_REGIONS with CFG_MMAP_REGIONS is simpler. How about the following patch?
```
diff --git a/core/arch/arm/arm.mk b/core/arch/arm/arm.mk
index bb65f77..6e52deb 100644
--- a/core/arch/arm/arm.mk
+++ b/core/arch/arm/arm.mk
@@ -4,6 +4,8 @@ CFG_LTC_OPTEE_THREAD ?= y
 CFG_CORE_TZSRAM_EMUL_SIZE ?= 458752
 CFG_LPAE_ADDR_SPACE_SIZE ?= (1ull << 32)
 
+CFG_MMAP_REGIONS ?= 13
+
 ifeq ($(CFG_ARM64_core),y)
 CFG_KERN_LINKER_FORMAT ?= elf64-littleaarch64
 CFG_KERN_LINKER_ARCH ?= aarch64
diff --git a/core/arch/arm/mm/core_mmu.c b/core/arch/arm/mm/core_mmu.c
index f2d728d..7c5fb3d 100644
--- a/core/arch/arm/mm/core_mmu.c
+++ b/core/arch/arm/mm/core_mmu.c
@@ -51,7 +51,6 @@
 
 #include "core_mmu_private.h"
 
-#define MAX_MMAP_REGIONS       13
 #define RES_VASPACE_SIZE       (CORE_MMU_PGDIR_SIZE * 10)
 #define SHM_VASPACE_SIZE       (1024 * 1024 * 32)
 
@@ -66,7 +65,7 @@ unsigned long default_nsec_shm_size;
 unsigned long default_nsec_shm_paddr;
 
 static struct tee_mmap_region
-       static_memory_map[MAX_MMAP_REGIONS + 1];
+       static_memory_map[CFG_MMAP_REGIONS + 1];
 
 /* Define the platform's memory layout. */
 struct memaccess_area {
@@ -911,6 +910,7 @@ void core_init_mmu_map(void)
                        panic("Invalid memory access config: sec/nsec");
        }
 
+       COMPILE_TIME_ASSERT(CFG_MMAP_REGIONS < 13);
        init_mem_map(static_memory_map, ARRAY_SIZE(static_memory_map));
 
        map = static_memory_map;
```

Reply:
@MrVan looks good to me!

Reply:
I think `MAX_MMAP_REGION` does not need to reach make script.
This value allocated static memory so it's a good thing to have it tunable. But the amount of memory is quite low. Maybe we would need a debug trace trace to get the number of entry used against max defined value.

In this scope, we should rather look at `MAX_XLAT_TABLES` which cover bigger areas.

Minor comment: why restrict to 13. If no valid value, will panic. Ok 13 to be a default value, but why the lowest supported ?

Reply:
@etienne-lms I have no idea about the default value 13, 13 is not large enough in my case. The patch I listed above is the easy way to fix the issue I met.

Covering bigger areas, do you mean `XLAT_TABLE_ENTRIES`?